plugins/inventory/azure_rm.py

```diff
@@ -584,6 +610,69 @@ def _legacy_script_compatible_group_sanitization(name):
 # VMSS VMs (all SS, N specific SS, N resource groups?): SS -> VM -> InstanceView, N NICs, N PublicIPAddress)
 
 
+class ArcHost(object):
+    def __init__(self, arc_model, inventory_client, legacy_name=False):
+        self._inventory_client = inventory_client
+        self._arc_model = arc_model
+        self._instanceview = self._arc_model
+        self._status = self._arc_model['properties'].get('status', {}).lower()  # 'Connected'
+        self._powerstate = self._status.replace('connected', 'running')
+
+        self._hostvars = {}
+
+        arc_name = self._arc_model['name']
+
+        if legacy_name:
+            self.default_inventory_hostname = arc_name
+        else:
+            # Azure often doesn't provide a globally-unique filename, so use resource name + a chunk of ID hash
+            self.default_inventory_hostname = '{0}_{1}'.format(arc_name, hashlib.sha1(to_bytes(arc_model['id'])).hexdigest()[0:4])
+
+    @property
+    def hostvars(self):
+        if self._hostvars != {}:
+            return self._hostvars
+
+        properties = self._arc_model.get('properties', {})
+        new_hostvars = dict(
+            network_interface=[],
+            mac_address=[],
+            ansible_all_ipv4_addresses=[],
+            ansible_all_ipv6_addresses=[],
+            public_ipv4_address=[],
+            private_ipv4_addresses=[],
+            public_dns_hostnames=[],
+            ansible_dns=[],
+            id=self._arc_model['id'],
+            location=self._arc_model['location'],
+            name=self._arc_model['name'],
+            default_inventory_hostname=self.default_inventory_hostname,
+            powerstate=self._powerstate,
+            status=self._status,
+            provisioning_state=properties.get('provisioningState', 'unknown').lower(),
+            os_profile=dict(
+                sku=properties.get('osSku', 'unknown'),
+                system=properties.get('osType', 'unknown'),
+                version=properties.get('osVersion', 'unknown'),
+            ),
+            tags=self._arc_model.get('tags', {}),
+            resource_type=self._arc_model.get('type', "unknown"),
+            resource_group=parse_resource_id(self._arc_model['id']).get('resource_group').lower(),
+        )
+
+        for nic in properties.get('networkProfile', {}).get('networkInterfaces', []):
+            new_hostvars['mac_address'].append(nic.get('macAddress'))
+            new_hostvars['network_interface'].append(nic.get('name'))
+            for ipaddr in nic.get('ipAddresses', []):
+                ipAddressVersion = ipaddr.get('ipAddressVersion')
+                if ipAddressVersion == 'IPv4':
+                    new_hostvars['ansible_all_ipv4_addresses'].append(ipaddr.get('address'))
```

Comment:
also fill  private_ipv4_addresses ?

Reply:
I included the netaddr package and use is_global() to determine